### PR TITLE
[#4022] Throwing Generic Exceptions - Adaptive Expressions

### PR DESCRIPTION
**Pull request by ceciliaavila**
Addresses #4022

## Description

This PR replaces all the generic expressions in `AdaptiveExpressions`, `Adapters`, `integration`, `FunctionalTests`, and part of `BotBuilder` libraries. Also, updates the affected tests to assert the proper exception.

## Specific Changes
Replaced generic exceptions in:
- FunctionalTests
- libraries\Adapters
- libraries\AdaptiveExpressions
- libraries\integration
- libraries\Microsoft.Bot.Builder\Adapters

Updated tests to assert proper exception:
- tests\Microsoft.Bot.Builder.Tests\Adapters\TestFlowTests.cs

## Testing
image

### CODE REVIEW

**Review by boydc2014 — APPROVED** [review]

**Comment by boydc2014** [comment]
Thanks @ceciliaavila, looks good. I'm not sure invalidOperationException is the perfect one for some syntax error of exception, but i can't find a better one, so i guess it's OK.

**Review by gabog — APPROVED** [review]
I agree that InvalidOperationException is ambiguous but no more than just Exception, let's go with that one.

Also, not related to the issue that trigger this PR but I flagged a method where we call GetText() 5 times, this may be a perf issue and may need a separate issue for it.

File: libraries/AdaptiveExpressions/parser/ExpressionParser.cs [hunk 1]
```diff
@@ -196,7 +196,7 @@ public override Expression VisitNumericAtom([NotNull] ExpressionAntlrParser.Nume
                     return Expression.ConstantExpression(doubleValue);
                 }
 
-                throw new Exception($"{context.GetText()} is not a number in expression '{context.GetText()}'");
+                throw new ArgumentException($"{context.GetText()} is not a number in expression \"{context.GetText()}\"");
```
[the remark this answers is not in the document]
Comment (ceciliaavila) [reply libraries/AdaptiveExpressions/parser/ExpressionParser.cs:199]: Sure, we can fix that in this PR.

**Review by mrivera-ms — APPROVED** [review]

File: libraries/AdaptiveExpressions/parser/ExpressionParser.cs (hunk 1 above)
[the remark this answers is not in the document]
Comment (ceciliaavila) [reply libraries/AdaptiveExpressions/parser/ExpressionParser.cs]: Done @gabog. Thanks!

**Review by gabog — APPROVED** [review]

**Pull request merged**